File: 编程入门/C++基础/基础语法/基础语法/MyDog.cpp
#include "MyDog.h"

MyDog::MyDog(const string& arg_breed, const string& arg_name, bool arg_master) :breed(arg_breed), name(arg_name), master(arg_master)
{
	cout << "我的品种是: " << breed << endl;
	cout << "我的名字是: " << name << endl;
	if (master)
		cout << "我有主人，不是流浪狗" << endl;
	else
		cout << "我没有主人，是一条流浪狗" << endl;
}
// 狂叫能力
void MyDog::bark(void)
{
	cout << "无能狂怒..." << endl;
}

// 跳跃能力
void MyDog::jump(void)
{
	cout << "跳跃..." << endl;
}

// 拆家能力
void MyDog::demolition(void)
{
	cout << "拆家..." << endl;
}

void MyDog::setName(const string& name)
{
	this->name = name;
}

string MyDog::getName(void)
{
	return this->name;
}

void MyDog::setBreed(const string& breed)
{
	this->breed = breed;
}

string MyDog::getBreed(void)
{
	return this->breed;
}

void MyDog::setMaster(bool master)
{
	this->master = master;
}

bool MyDog::getMaster(void)
{
	return this->master;
}

MyDog::~MyDog()
{
	cout << "主人来世再见..." << endl;
}

File: 编程入门/java基础/javas/src/com/study/oob/MyDog.java
package com.study.oob;

public class MyDog {

    private String breed; // 品种
    private String name; // 名字
    private boolean master; // 是否有主人

    public MyDog(String breed, String name, boolean master) {
        this.breed = breed;
        this.name = name;
        this.master = master;
    }

    public String getBreed() {
        return breed;
    }

    public void setBreed(String breed) {
        this.breed = breed;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public boolean isMaster() {
        return master;
    }

    public void setMaster(boolean master) {
        this.master = master;
    }

    // 狂叫能力
    public void bark() {
        System.out.println("无能狂怒...");
    }

    // 跳跃能力
    public void jump() {
        System.out.println("跳跃...");
    }

    // 拆家能力
    public void demolition() {
        System.out.println("拆家...");
    }

    @Override
    public String toString() {
        return "MyDog{" +
                "breed='" + breed + '\'' +
                ", name='" + name + '\'' +
                ", master=" + master +
                '}';
    }
}
